### PR TITLE
Feature/homework 8 router

### PR DESCRIPTION
**Pull request by bmitioglov**

### CODE REVIEW

File: src/containers/MoviesList/movieslist.js [hunk 1]
```diff
@@ -13,17 +14,28 @@ const MoviesList = ({ onMovieClick }) => {
...
   });
   
+  const location = useLocation();
+  
+  const urlValue = location.search;
+  const searchValue = new URLSearchParams(urlValue).get('search');
+  
+  const category = useSelector((state) => selectCategory(state));
+  
+  //TODO: fix
```
Comment (endiny) [inline src/containers/MoviesList/movieslist.js]: fix what? :)

File: src/components/Search/search.js [hunk 2]
```diff
@@ -1,14 +1,36 @@
...
+  }
+  
+  return (
+    <div className="search">
+      <h3 className="SearchHeader">Find your movie</h3>
+      <div className="search-bar">
+        <input className="search-input" type="text"
+               onKeyPress={enterPressed.bind(this)}
+               onChange={(event) => setSearchInput(event.target.value)} />
+        <button className="search-button" type="button" onClick={handleClick}>Search</button>
```
Comment (endiny) [inline src/components/Search/search.js:30]: nit: technically that would be a link navigating to the search results, semantically still a button
https://reactrouter.com/web/api/Link

File: src/components/MovieCard/moviecard.js [hunk 3]
```diff
@@ -34,10 +36,12 @@ const MovieCard = ({id, title, description, year, image, releaseDate, runtime, g
     { title: 'Delete', callback: toggleDelete },
   ];
   
+  
   return (
     <div className="movie-card">
       <div className="image-container">
-        <img onClick={onClick} className="movie-image" src={image} alt={title} />
+        <img onClick={function(event){ onClick(); history.push(`/movie/${id}`); }}
```
Comment (endiny) [inline src/components/MovieCard/moviecard.js]: I'd suggest to extract toggleMovieHeader to redux as well, so you can make it a https://reactrouter.com/web/api/Link

File: src/components/Search/search.js [hunk 4]
```diff
@@ -1,14 +1,36 @@
...
+      handleClick();
+    }
+  }
+  
+  return (
+    <div className="search">
+      <h3 className="SearchHeader">Find your movie</h3>
+      <div className="search-bar">
+        <input className="search-input" type="text"
+               onKeyPress={enterPressed.bind(this)}
```
Comment (endiny) [inline src/components/Search/search.js]: Why are you binding the function? Arrow function has no own context preserving the parent one. So basically you would bind the function to the module context. Still you ain't need it

File: src/components/Search/search.js [hunk 5]
```diff
@@ -1,14 +1,36 @@
...
+  const history = useHistory();
+  
+  function handleClick() {
+    if (searchInput !== '') {
+      history.push(`/?search=${searchInput}`);
+    }
+  }
+  
+  function enterPressed(event) {
+    let code = event.keyCode || event.which;
```
Comment (endiny) [inline src/components/Search/search.js:17]: React produces own intercompatible synthetic events. `keyCode` would be enough

File: src/components/Search/search.js [hunk 6]
```diff
@@ -1,14 +1,36 @@
...
+import { useHistory } from 'react-router-dom';
 
-const Search = () => (
-  <div className="search">
-    <h3 className="SearchHeader">Find your movie</h3>
-    <div className="search-bar">
-      <input className="search-input" type="text" />
-      <button className="search-button" type="button">Search</button>
+const Search = () => {
+  const [searchInput, setSearchInput] = useState('');
```
Comment (endiny) [inline src/components/Search/search.js:6]: nit: I believe keeping it in redux would suit better.

File: src/components/notfoundpage/notfoundpage.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  
+  function handleClick() {
+    history.push('/');
+  }
+  
+  return (
+    <div className="error-container">
+      <div className="error-label">Page not found</div>
+      <div className="error-label">404</div>
+      <button className="go-back-button" type="button" onClick={handleClick}>Go back to Home</button>
```
Comment (endiny) [inline src/components/notfoundpage/notfoundpage.js]: https://reactrouter.com/web/api/Link

File: src/containers/MoviesList/movieslist.js [hunk 8]
```diff
@@ -13,17 +14,28 @@ const MoviesList = ({ onMovieClick }) => {
     return state.movies.status;
   });
   
+  const location = useLocation();
+  
+  const urlValue = location.search;
+  const searchValue = new URLSearchParams(urlValue).get('search');
+  
+  const category = useSelector((state) => selectCategory(state));
```
Comment (endiny) [inline src/containers/MoviesList/movieslist.js]: `useSelector(selectCategory)`?

**Review by endiny — CHANGES_REQUESTED** [review]

**Comment by endiny** [comment]
Please, rename all js files with components to `.jsx`

File: src/containers/MoviesList/movieslist.js (hunk 1 above)
Comment (bmitioglov) [reply src/containers/MoviesList/movieslist.js]: by some reason this useEffect executes 2 times, I don't know why...

File: src/components/MovieCard/moviecard.js (hunk 3 above)
Comment (bmitioglov) [reply src/components/MovieCard/moviecard.js]: done

File: src/components/Search/search.js (hunk 4 above)
Comment (bmitioglov) [reply src/components/Search/search.js]: fixed

File: src/components/notfoundpage/notfoundpage.js (hunk 7 above)
Comment (bmitioglov) [reply src/components/notfoundpage/notfoundpage.js]: fixed

File: src/containers/MoviesList/movieslist.js (hunk 8 above)
Comment (bmitioglov) [reply src/containers/MoviesList/movieslist.js]: fixed

**Pull request merged**